### ISSUE TITLE
A few issues with `oktaAuth.handleLoginRedirect()`

### ISSUE THREAD

**Issue by estorski** [okta/okta-react#77]
Hey folks,

We have an issue in the up to date version of `@okta` libs (`4.1.0` for `@okta/okta-react` and `4.6.1` for `@okta/okta-auth-js`).
We use OKTA-hosted login page.

The flow in question is when during login procedure the user does not allow access to a claim (`user-data:read` in our case), the user gets redirected back to the application login-callback page. On that said page we have code like this:
```
  const { authState, oktaAuth } = useOktaAuth();
  const [error, setError] = useState<OAuthError | undefined>();

  useEffect(() => {
    if (authState.isPending) {
      void oktaAuth.handleLoginRedirect().catch(ex => setError(ex));
    }
  }, [authState.isPending, oktaAuth]);
```
And there are two issues with this:
1. The `authState` never gets updated with the `error`, that is why we had to `.catch()` it via Promise syntax and save it to state in order to show it on the page. Normally I'd expect `authState.error` to appear, but that never happens in practice.
2. After that said error happened, the `authState` object is stuck with `isPending === true` and `isAuthenticated === false` and we have no clue on how to get it out of this state without reloading the whole SPA.

I'd appreciate any ideas on the above.

**Comment by oleksandrpravosudko-okta** [comment]
Thanks for report @estorski.

I'll create a ticket to investigate this:
Internal ref: OKTA-362651

To confirm we are using correct repro setup: are you requesting [user consent](https://developer.okta.com/docs/guides/request-user-consent/overview/) during login?

**Comment by estorski** [comment]
Yes, that's the user consent request that gets rejected by the user

**Comment by holdtb** [comment]
Dealing with something similar myself when trying to sign in as a user that hasn't been assigned the client application. I am able to repro using both the provided LoginCallback component, and a custom LoginCallback component. 

`oktaAuth.handleLoginRedirect()` causes an exception (Unhandled Rejection (OAuthError): User is not assigned to the client application.) to be thrown that appears to be unhandled by `@okta/okta-react`. If I wrap the login call with a catch like @estorski did above, then I see the same end result. The app no longer crashes, but the authState object get's stuck with `isPending === true` and `isAuthenticated === false` and other portions of the app begin to misbehave. The error is not set on the authState object and perhaps most importantly, the ErrorComponent of the LoginCallback is never rendered.

"@okta/okta-auth-js": 4.6.1
"@okta/okta-react": 4.1.0

**Comment by ckifer** [comment]
Plus one to this, especially @holdtb 's comment above.  `LoginCallback` never renders the error component. Same versions.

**Comment by swiftone** [comment]
We have a task to work on a fix this sprint - I cannot promise results until we have them, but I can state that we are actively attacking this issue.

**Comment by erlioniel** [comment]
@swiftone I think it's partially related to https://github.com/okta/okta-auth-js/issues/488 (at least the first part)

**Comment by darkowic** [comment]
Same issue here with using directly `oktaAuth.handleLoginRedirect();`. I would expect that this is internally handled and returned as auth state change since it is part of OktaAuth class.

Workaround...
```tsx
export const LoginCallback: React.FC<LoginCallbackProps> = ({
    errorComponent: ErrorComponent,
}) => {
    const [loginError, setLoginError] = useState<Error>();
    const { error, isPending } = useContext(AuthContext);
    const { authService } = useAuthState();

    useEffect(() => {
        let hasUnmounted = false;
        const handleLoginRedirect = async () => {
            try {
                await authService.handleLoginRedirect();
            } catch (e) {
                if (!hasUnmounted) {
                    setLoginError(e);
                }
            }
        };

        handleLoginRedirect();

        return () => {
            hasUnmounted = true;
        };
    }, [authService]);

    const finalError = loginError || (!isPending && error);

    if (finalError) {
        return <ErrorComponent error={finalError} />;
    }

    return null;
};
```

**Comment by aarongranick-okta** [comment]
@estorski @darkowic @erlioniel @ckifer We have released a fix for this issue. After upgrading to `@okta/okta-react` [version 5.1.1](https://github.com/okta/okta-react/releases/tag/okta-react-5.1.1) you should see OAuth callback errors caught and displayed in the `LoginCallback` component. Please let us know if this has resolved your issue, or if you are still seeing a problem.

**Comment by denysoblohin-okta** [comment]
Closing this issue as it was fixed in 5.1.1
